plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -192,9 +191,8 @@ makeLenses ''TypeCheckEnv
 type TypeCheckT uni fun cfg m = ReaderT (TypeCheckEnv uni fun cfg) m
 
 -- | The constraints that are required for kind checking.
-type MonadKindCheck err term uni fun ann m =
-    ( MonadError err m                  -- Kind/type checking can fail
-    , AsTypeError err term uni fun ann  -- with a 'TypeError'.
+type MonadKindCheck error term uni fun ann m =
+    ( MonadError error m                -- Kind/type checking can fail
```

Comment:
At this point is there even benefit to keeping `MonadError` here? Can it be just `Either`? Can we simply move this functionality into `TypeCheckT`?

Anyway, that's probably is best to investigate separately, so feel free to ignore it here.

Reply:
I'd be happy to kill MonadError as well. It's probably best in a separate PR however. I'll make an issue.